core/src/main/kotlin/com/strumenta/kolasu/model/Naming.kt

```diff
@@ -22,21 +25,31 @@ interface Named : PossiblyNamed {
     override val name: String
 }
 
+data class ReferenceChangeNotification<N : PossiblyNamed>(val oldValue: N?, val newValue: N?)
+
 /**
  * A reference associated by using a name.
  * It can be used only to refer to Nodes and not to other values.
  *
  * This is not statically enforced as we may want to use some interface, which cannot extend Node.
  * However, this is enforced dynamically.
  */
-class ReferenceByName<N>(val name: String, initialReferred: N? = null) : Serializable where N : PossiblyNamed {
+class ReferenceByName<N>(val name: String, initialReferred: N? = null) :
+    Serializable,
+    Disposable,
+    ObservableSource<ReferenceChangeNotification<N>> where N : PossiblyNamed {
+
+    private val observers: MutableList<Observer<in ReferenceChangeNotification<N>>> = mutableListOf()
```

Comment:
Usually in situations like this I've employed a [Subject](https://github.com/ReactiveX/RxJava/wiki/Subject) instead of managing a list of observers manually.
Either a `BehaviorSubject` (if you want new observers to be notified of the last changed value upon subscription) or a `PublishSubject` (if you want new observers to only see changes after they've subscribed).

In Angular, the convention is to append a dollar sign to the subject, e.g.
```
val changes$ = PublishSubject<...>()
```
but I don't know if that's an Angular-specific convention or if it's common across Rx* libraries.

Either way, subscribers would access the public property e.g.

```
myReference.changes.subscribe(...)
```

Then, you don't have to manage the observers and call `onNext` on each, you can just call `onNext` on the Subject. See also https://github.com/ManyDesigns/Portofino/blob/deaba48d93502c1610b830781fa78855e6e05160/portofino-database/src/main/java/com/manydesigns/portofino/persistence/Persistence.java#LL106C45-L106C45 for an example using RxJava.

Reply:
Implemented, please let me know if I did correctly